src/structures/interfaces/Collector.js

```diff
@@ -95,6 +96,29 @@ class Collector extends EventEmitter {
     if (post) this.stop(post);
   }
 
+  /**
+   * Call this to remove an element from the collection. Accepts any event data as parameters.
+   * @param {...*} args The arguments emitted by the listener
+   * @emits Collector#remove
+   * @private
+   */
+  _handleRemove(...args) {
+    const remove = this.remove(...args);
+    if (!remove || !this.filter(...args, this.collected)) return;
+    this.collected.delete(remove.key);
+
+    /**
+     * Emitted whenever an element is removed.
+     * @event Collector#remove
+     * @param {*} element The element that got removed.
+     * @param {Collector} collector The collector
+     */
+    this.emit('remove', remove.value, this);
```

Comment:
I would suggest renaming this event to `dispose` to keep consistent with the next version of Discord.js, since this functionality was added to master in #1616 with a different event name.